src/interface/lazy.jl

```diff
@@ -189,6 +191,8 @@ function Base.copyto!(dst::AbstractArray, src::LazyTensor{T, N}) where {T, N}
     return LazyTensor{T, N}(reformat(immediate(dst), src.data), src.extrude)
 end
 
+Base.permutedims(swizzle_arr::SwizzleArray{dims, <:Tensor}, perm) where {dims} =
+    compute(permutedims(permutedims(LazyTensor(swizzle_arr.body), dims), perm))
```

Comment:
These two lines can just be

`Base.permutedims(arr::SwizzleArray, perm) = swizzle(arr, perm...)`

Reply:
Ah, right - done!